Apps/Examples/Examples/All Examples/AdvancedViewportGesturesExample.swift

```diff
@@ -0,0 +1,53 @@
+import UIKit
+import MapboxMaps
+
+@objc(AdvancedViewportGesturesExample)
+final class AdvancedViewportGesturesExample: UIViewController, ExampleProtocol {
+    private var mapView: MapView!
+    private var followingViewportState: FollowingViewportState!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        mapView = MapView(frame: view.bounds)
+        view.addSubview(mapView)
+
+        mapView.location.options.puckType = .puck2D()
+
+        followingViewportState = mapView.viewport.makeFollowingViewportState(
+            options: FollowingViewportStateOptions(bearing: .course))
+
+        mapView.viewport.transition(to: followingViewportState)
+
+        mapView.gestures.options.panEnabled = false
+        mapView.gestures.options.pitchEnabled = false
+        mapView.gestures.options.pinchEnabled = false
+        mapView.gestures.options.doubleTapToZoomInEnabled = false
+        mapView.gestures.options.doubleTouchToZoomOutEnabled = false
+        mapView.gestures.options.quickZoomEnabled = false
+        mapView.gestures.options.animationLockoutEnabled = false
+
+        let doubleTapGestureRecognzier = UITapGestureRecognizer(target: self, action: #selector(zoomIn))
+        doubleTapGestureRecognzier.numberOfTapsRequired = 2
+        doubleTapGestureRecognzier.numberOfTouchesRequired = 1
+        mapView.addGestureRecognizer(doubleTapGestureRecognzier)
+
+        let doubleTouchGestureRecognzier = UITapGestureRecognizer(target: self, action: #selector(zoomOut))
+        doubleTouchGestureRecognzier.numberOfTapsRequired = 1
+        doubleTouchGestureRecognzier.numberOfTouchesRequired = 2
+        mapView.addGestureRecognizer(doubleTouchGestureRecognzier)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+         // The below line is used for internal testing purposes only.
+        finish()
+    }
+
+    @objc private func zoomIn() {
+        followingViewportState.options.zoom += 1
+    }
+
+    @objc private func zoomOut() {
+        followingViewportState.options.zoom -= 1
+    }
+}
```

Comment:
@pengdev @LukasPaczos here's a quick demo of doing more advanced gestures using existing viewport API. I did add one gesture options to allow disabling the behavior where touching the map cancels any ongoing animations. We'll need to think about that a bit more carefully.

I imagine there are some things we could do to make this a bit nicer. I'll continue thinking about it.

Reply:
Removed from this PR for now. Will revisit in a future PR.

Reply:
I added an AdvanedViewportGesturesActivity to the Android PR as well